client/providers/google.go

```diff
@@ -120,12 +120,19 @@ func (g *GoogleOp) VerifyCICHash(ctx context.Context, idt []byte, expectedCICHas
 	return nil
 }
 
-func (g *GoogleOp) PublicKey(ctx context.Context, idt []byte) (crypto.PublicKey, error) {
-	j, err := jws.Parse(idt)
+func (g *GoogleOp) PublicKey(ctx context.Context, headersEnc []byte) (crypto.PublicKey, error) {
+	headersJSON, err := util.Base64DecodeForJWT(headersEnc)
 	if err != nil {
-		return nil, fmt.Errorf("failed to parse JWS: %w", err)
+		return nil, err
+	}
+
+	headers := make(map[string]any)
```

Comment:
Why would the headers not be `map[string]string`? Is something unmarshaling the strings into something else?

Reply:
You're right, this is just me being lazy

Reply:
I don't know if forcing yourself to cast to string every time you need to use it counts as laziness.

Reply:
I originally changed this to unmarshal to a struct, but on second thought, I think the map[string]any is better. This is untrusted input so it really could be anything, and when unmarshalling into a struct, if the key is missing we'll just get an empty string. With a map we can check it's there and the correct type.